fsspec/implementations/sftp.py

```diff
@@ -64,9 +64,20 @@ def _get_kwargs_from_urls(urlpath):
         out.pop("protocol", None)
         return out
 
-    def mkdir(self, path, mode=511):
-        logger.debug("Creating folder %s" % path)
-        self.ftp.mkdir(path, mode)
+    def mkdir(self, path, create_parents=False, mode=511):
+        logger.info("Creating folder %s" % path)
+        if self.exists(path):
+            raise FileExistsError("File exists: {}".format(path))
+
+        if create_parents:
+            parts = path.split("/")
+            path = ""
+            for part in parts:
+                path += "/" + part
+                if not self.exists(path):
+                    self.ftp.mkdir(path, mode)
```

Comment:
If we find a path that exists, we should stop the loop and not keep checking the parents (which must therefore also exist)

Reply:
We will not know how parents will a directory has, that's why it checks and creates according to a parent does not exist. 

In a condition for mkdir('/a/b/c') usage, such that `create_parents=True`  and '/a/b/' exists, it will loop extra as you pointed out. But we still have no information where to split (meaning which part exists).

Reply:
If '/a/b/c' exists, then so must '/a/b', we do not need to check

Reply:
Yes, that would be better